packages/block-editor/src/components/height-control/index.js

```diff
@@ -95,13 +100,14 @@ export default function HeightControl( {
 	};
 
 	return (
-		<fieldset className="block-editor-height-control">
-			<BaseControl.VisualLabel as="legend">
+		<div className="block-editor-height-control" id={ id }>
+			<BaseControl.VisualLabel as="label" for={ inputId } id={ labelId }>
```

Comment:
Reading at the doc, it looks like `BaseControl.VisualLabel` is not meant to be used to render an actual `label` (even though it's possible to do so, through the `as` prop) — from the [docs](https://github.com/WordPress/gutenberg/blob/64a873a80b001a1c2c892c0c4a56010a94cc4c03/packages/components/src/base-control/README.md#basecontrolvisuallabel):

> `BaseControl.VisualLabel` is used to render a purely visual label inside a `BaseControl` component.
>
> It should only be used in cases where the children being rendered inside BaseControl are already accessibly labeled, e.g., a button, but we want an additional visual label for that section equivalent to the labels `BaseControl` would otherwise use if the `label` prop was passed.

I wonder if:
- we should keep the current solution (which means that we should probably update the `BaseControl.VisualLabel` docs?)
- we should render an additional `<label />` element (and make it visually hidden). At the point, we could remove the `as` prop on `BaseControl.VisualLabel` and use it only for aesthetical reasons
- remove `BaseControl.VisualLabel` and replace it with a simple `<label />` (although we'd need to somewhat style it).

Maybe @mirka has some more insights on what would be the better approach?

Reply:
I'd like to see how we could enhance the `BaseControl` component itself so that it can support this kind of labeling pattern.

Reply:
I'd need to spend more time looking at how `BaseControl` works to offer much feedback around any of this!